allennlp/layers/embeddings.py

```diff
@@ -0,0 +1,173 @@
+import codecs
+import gzip
+import logging
+
+import numpy
+import torch
+
+from ..data.vocabulary import Vocabulary
+from ..common.checks import ConfigurationError
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Embedding(torch.nn.Module):
+    """
+    A more featureful embedding module than the default in Pytorch.
+    Adds the ability to pre-specify the weight matrix, use a non-trainable embedding and
+    catches some ill-advised use cases, such as 1 dimensional embeddings.
+
+    Parameters
+    ----------
+    num_embeddings:, int:
```

Comment:
We should probably figure out markdown vs. rest sooner rather than later, so we can switch docstrings now, if we're going to switch.

If we're sticking with rest, this should be formatted like:
```
num_embeddings : ``int``
```
Note the space after the parameter name - it's important for the code that generates the docs.